Added SendResetPasswordEmail method

vellenge opened the pull request:
Added SendResetPasswordEmail method from Firebase Auth to EzLogin. This new method returns a nullable string instead of an EzLoginStatus to avoid modification and misusage of this enum and to use the returned error code.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cr-crme/ezlogin/6)
<!-- Reviewable:end -->

vellenge committed:
Added SendResetPasswordEmail method

Added SendResetPasswordEmail method from Firebase Auth to EzLogin. This new method returns a nullable string instead of an EzLoginStatus to avoid modification and misusage of this enum

vellenge committed:
Added SendResetPasswordEmail method

Added SendResetPasswordEmail method from Firebase Auth to EzLogin. This new method returns a nullable string instead of an EzLoginStatus to avoid modification and misusage of this enum and to use the returned error code.

vellenge committed:
Merge remote-tracking branch 'vellenge/main' into password-recovery

pariterre reviewed (changes requested):
> @pariterre reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cr-crme/ezlogin/6)* status: all files reviewed, 4 unresolved discussions (waiting on @vellenge)

___
*[`lib/ezlogin_firebase.dart` line 238 at r1](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObNtoj9EYmFvJF251b4:-ObNtoj9EYmFvJF251b5:b3reik4) ([raw file](https://github.com/cr-crme/ezlogin/blob/bc7ae2e22ede043200c1a6be39d7b2a0d5f0629d/lib/ezlogin_firebase.dart#L238)):*
> ```Dart
>   }
> 
>   Future<String?> sendPasswordResetEmail({required String email}) async {
> ```

I agree that it should not return any string, however it should return a bool "success" or "fail"
___
*[`lib/ezlogin_firebase.dart` line 238 at r1](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObNu5vMBll7Uvv_Juqv:-ObNu5vMBll7Uvv_Juqw:b-vgep5q) ([raw file](https://github.com/cr-crme/ezlogin/blob/bc7ae2e22ede043200c1a6be39d7b2a0d5f0629d/lib/ezlogin_firebase.dart#L238)):*
> ```Dart
>   }
> 
>   Future<String?> sendPasswordResetEmail({required String email}) async {
> ```

the method should be called "resetPassword" and should be added to Ezlogin as an abstract method (so all the implementation are force to provide a reset password method (or fail with a raise)
___
*[`lib/ezlogin_firebase.dart` line 243 at r1](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObNu3-k9fzJlnQejN59:-ObNu3-k9fzJlnQejN5A:bi73kxk) ([raw file](https://github.com/cr-crme/ezlogin/blob/bc7ae2e22ede043200c1a6be39d7b2a0d5f0629d/lib/ezlogin_firebase.dart#L243)):*
> ```Dart
>     } on FirebaseAuthException catch (e) {
>       debugPrint(e.code);
>       return e.code.toString();
> ```

return false
___
*[`lib/ezlogin_firebase.dart` line 245 at r1](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObNu5FaBgvHBGC7uc7B:-ObNu5FaBgvHBGC7uc7C:bpsr3vd) ([raw file](https://github.com/cr-crme/ezlogin/blob/bc7ae2e22ede043200c1a6be39d7b2a0d5f0629d/lib/ezlogin_firebase.dart#L245)):*
> ```Dart
>       return e.code.toString();
>     }
>     return null;
> ```

return true


<!-- Sent from Reviewable.io -->

vellenge committed:
Updated SenResetPasswordEmail to resetPassword

Changed resetPassword methods to return a boolean instead of a nullable string.
Made resetPassword in ezlogin_firebase an override of a super resetPassword method in ezlogin
Added an unimplemented resetPassword override in ezlogin_mock, which throws.

pariterre reviewed (changes requested):
> @pariterre reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cr-crme/ezlogin/6)* status: all files reviewed, 3 unresolved discussions (waiting on @vellenge)

___
*[`lib/ezlogin_mock.dart` line 128 at r2](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObU1S928IvgZya7jHZg:-ObU1S928IvgZya7jHZh:b-kxu7v0) ([raw file](https://github.com/cr-crme/ezlogin/blob/e0ff49e9089d68841501c940128254a887b20b44/lib/ezlogin_mock.dart#L128)):*
> ```Dart
>   @override
>   Future<bool> resetPassword({required String email}) {
>     // TODO: implement resetPassword
> ```

Remove this comment
___
*[`lib/ezlogin_mock.dart` line 129 at r2](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObU1_fO911OipxKdxvi:-ObU1_fO911OipxKdxvj:b-tztwuh) ([raw file](https://github.com/cr-crme/ezlogin/blob/e0ff49e9089d68841501c940128254a887b20b44/lib/ezlogin_mock.dart#L129)):*
> ```Dart
>   Future<bool> resetPassword({required String email}) {
>     // TODO: implement resetPassword
>     throw UnimplementedError();
> ```

This mock can return "true" instead of throwing (it is a mock after all!)
___
*[`lib/ezlogin.dart` line 140 at r2](https://reviewable.io/reviews/cr-crme/ezlogin/6#-ObU1HV6349GpZCNzWyC:-ObU1HV6349GpZCNzWyD:b-v7tv0q) ([raw file](https://github.com/cr-crme/ezlogin/blob/e0ff49e9089d68841501c940128254a887b20b44/lib/ezlogin.dart#L140)):*
> ```Dart
>   Future<EzloginStatus> clearUsers();
> 
>   Future<bool> resetPassword({required String email});
> ```

Please add a docstring (see other methods)


<!-- Sent from Reviewable.io -->

vellenge committed:
Updated resetPassword methods

Added docstring for resetPassword in ezlogin
Added default return value for resetPassword in ezlogin_mock

pariterre reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @pariterre reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cr-crme/ezlogin/6#-:-ObXidrEADWuOfpuyotE:bnfp4nl)* status: all files reviewed, 3 unresolved discussions (waiting on @vellenge)



<!-- Sent from Reviewable.io -->

pariterre committed:
Update resetPassword method documentation
